homeassistant/components/rest/binary_sensor.py

```diff
@@ -149,24 +149,28 @@ def _update_from_rest_data(self) -> None:
             self._attr_is_on = False
             return
 
-        response = self.rest.data
+        response = self.rest.data_without_xml()
 
         raw_value = response
 
-        if self._value_template is not None:
+        if response is not None and self._value_template is not None:
             response = self._value_template.async_render_with_possible_json_value(
-                self.rest.data, False
+                response, False
             )
 
         try:
-            self._attr_is_on = bool(int(response))
+            if response is not None:
+                self._attr_is_on = bool(int(response))
+            else:
+                self._attr_is_on = False
```

Comment:
The reason this is not covered by the tests is that `` does not return `None` if the data is not valid XML, instead it returns the data as is:
https://github.com/home-assistant/core/blob/d4ecf30b6a2696433b20e7e6c93fe23c2fda693f/homeassistant/components/rest/data.py#L72-L90